pkg/objstore/s3/s3.go

```diff
@@ -34,14 +34,15 @@ const DirDelim = "/"
 
 // Config stores the configuration for s3 bucket.
 type Config struct {
-	Bucket        string     `yaml:"bucket"`
-	Endpoint      string     `yaml:"endpoint"`
-	AccessKey     string     `yaml:"access_key"`
-	Insecure      bool       `yaml:"insecure"`
-	SignatureV2   bool       `yaml:"signature_version2"`
-	SSEEncryption bool       `yaml:"encrypt_sse"`
-	SecretKey     string     `yaml:"secret_key"`
-	HTTPConfig    HTTPConfig `yaml:"http_config"`
+	Bucket          string            `yaml:"bucket"`
+	Endpoint        string            `yaml:"endpoint"`
+	AccessKey       string            `yaml:"access_key"`
+	Insecure        bool              `yaml:"insecure"`
+	SignatureV2     bool              `yaml:"signature_version2"`
+	SSEEncryption   bool              `yaml:"encrypt_sse"`
+	SecretKey       string            `yaml:"secret_key"`
+	PutUserMetadata map[string]string `yaml:"put_user_metadata"`
```

Comment:
To have this more generic could we move it onto `HTTPConfig` as `headers`? As we may end up having to do something similar for other providers?

Reply:
Sure, but this seems to be only for PUT.. Whould you do HTTPConfig and put this header in all call to this provider? Might do the work.,

Reply:
Maybe we should wait then till we get more use cases before refactoring